examples/svg_viewer/src/main.rs

```diff
@@ -0,0 +1,334 @@
+#[macro_use]
+extern crate gfx;
+extern crate gfx_window_glutin;
+extern crate gfx_device_gl;
+extern crate glutin;
+extern crate lyon;
+extern crate clap;
+extern crate svgparser;
+
+// use lyon::extra::rust_logo::build_logo_path;
+// use lyon::path_builder::*;
+use lyon::math::*;
+use lyon::tessellation::geometry_builder::{VertexConstructor, VertexBuffers, BuffersBuilder};
+use lyon::tessellation::{FillTessellator, FillOptions};
+use lyon::tessellation;
+use lyon::path::Path;
+use lyon::svg::parser::build_path;
+// use lyon::lyon_path_builder::SvgBuilder;
+
+use gfx::traits::{Device, FactoryExt};
+use glutin::GlContext;
+
+use clap::{Arg, App};
+
+use std::fs::File;
+use std::io::{Read};
+// use std::borrow::Borrow;
+
+use svgparser::svg::{Tokenizer, Token};
+use svgparser::{Tokenize, ElementId, AttributeId, Color};
+
+type ColorFormat = gfx::format::Rgba8;
+type DepthFormat = gfx::format::DepthStencil;
+
+gfx_defines!{
+    vertex GpuFillVertex {
+        position: [f32; 2] = "a_position",
+        color: [f32; 4] = "a_color",
+    }
+
+    constant Globals {
+        zoom: [f32; 2] = "u_zoom",
+        pan: [f32; 2] = "u_pan",
+    }
+
+    pipeline fill_pipeline {
+        vbo: gfx::VertexBuffer<GpuFillVertex> = (),
+        out_color: gfx::RenderTarget<ColorFormat> = "out_color",
+        constants: gfx::ConstantBuffer<Globals> = "Globals",
+    }
+}
+
+fn gpu_color(color: Color, opacity: f32) -> [f32; 4] {
+    [
+        f32::from(color.red) / 255.0,
+        f32::from(color.green) / 255.0,
+        f32::from(color.blue) / 255.0,
+        opacity
+    ]
+}
+
+// A very simple vertex constructor that only outputs the vertex position
+struct VertexCtor {
+    fill: Color
+}
+impl VertexConstructor<tessellation::FillVertex, GpuFillVertex> for VertexCtor {
+    fn new_vertex(&mut self, vertex: tessellation::FillVertex) -> GpuFillVertex {
+        assert!(!vertex.position.x.is_nan());
+        assert!(!vertex.position.y.is_nan());
+        GpuFillVertex {
+            // (ugly hack) tweak the vertext position so that the logo fits roughly
+            // within the (-1.0, 1.0) range.
+            position: (vertex.position * 0.0145 - vec2(1.0, 1.0)).to_array(),
```

Comment:
We should probably get rid of this hack here and use pan/zoom functionalities to navigate the viewer instead.